test/e2e/eventcounter/eventcounter_test.go

```diff
@@ -2001,26 +2189,28 @@ func getExperimentResult(t *testing.T, c ecclient.Client, experimentID string) *
 	}
 	var response *ecproto.GetExperimentResultResponse
 	var err error
-	numRetries := 6
+	numRetries := 3
 	for i := 0; i < numRetries; i++ {
 		response, err = c.GetExperimentResult(ctx, req)
 		if err == nil {
 			break
 		}
 		st, _ := status.FromError(err)
-		if st.Code() == codes.Internal {
-			t.Fatal(err)
+		if st.Code() != codes.Unavailable {
```

Comment:
I think we can also retry if the `st.Code()` is **Internal**. For example, we can return `InternalServerError` in the `GetExperimentResult` [function](https://github.com/bucketeer-io/bucketeer/blob/main/pkg/eventcounter/api/api.go#L761-L764), and this `InternalServerError` could also caused by downstream unavailable.

So this condition can be `st.Code() != codes.Unavailable && st.Code() != codes.Internal`.

`codes.Internal` Doc:
```golang
        // Internal errors. Means some invariants expected by underlying
	// system has been broken. If you see one of these errors,
	// something is very broken.
	//
	// This error code will be generated by the gRPC framework in several
	// internal error conditions.
        Internal Code = 13

```